bson-record-codec/src/main/org/bson/codecs/record/RecordCodec.java

```diff
@@ -275,6 +277,8 @@ public T decode(final BsonReader reader, final DecoderContext decoderContext) {
                 if (LOGGER.isTraceEnabled()) {
                     LOGGER.trace(format("Found property not present in the ClassModel: %s", fieldName));
                 }
+            } else if (reader.getCurrentBsonType() == BsonType.NULL && componentModel.isNullable) {
```

Comment:
One other change request, if you don't mind: I don't think this should fall through to the else branch if the BSON type is `NULL` but the component is not nullable.  Rather, it should throw an exception directly.  `BsonInvalidOperationException` is probably best.  Add a test for it too.

Reply:
Make sense - modified to let NULL on primitive field throw early and added one test case for it.